examples/postgres_payments/src/credit_card/aggregate.rs

```diff
@@ -58,8 +55,8 @@ impl Aggregate for CreditCardAggregate {
     }
 
     // No state for credit_card aggregate
-    fn apply_event(_id: &Uuid, state: CreditCardState, event: &StoreEvent<Self::Event>) -> CreditCardState {
-        match event.payload() {
+    fn apply_event(state: CreditCardState, event: &Self::Event) -> CreditCardState {
```

Comment:
I left the event wrapped inside the StoreEvent because I thought that maybe one of the inner field could have been useful for the state. Eg. assumptions over sequence_number or occurred_on. WDYT?

Reply:
I think that it makes sense for `projectors` etc to take the `StoreEvent` - since any metadata currently lives there.  (separately - I guess it could be worth thinking about how we handle metadata/tracing properly? (e.g. how pass through `correlation_id` through the system)

but it feels a bit odd to me for the core aggregate to have access to things like `occured_on`/`sequence_number` - depending on `sequence_number` in the state feels a bit flaky.  And as with `occured_on`, if the aggregate needs a sequence or a timestamp then potentially they should be placed in the commands/events explicitly?